app/controllers/coronavirus/reorder_sub_sections_controller.rb

```diff
@@ -35,10 +35,10 @@ def page
     end
 
     def set_positions
-      reordered_subsections = JSON.parse(params[:section_order_save])
-      reordered_subsections.each do |sub_section_data|
-        sub_section = page.sub_sections.find(sub_section_data["id"])
-        sub_section.update_column(:position, sub_section_data["position"])
+      reordered_subsections = params[:section_order_save]
+      reordered_subsections.each do |id, position|
+        sub_section = page.sub_sections.find(id)
+        sub_section.update_column(:position, position)
       end
```

Comment:
This is great - glad it doesn't seem tricky to integrate.

Reply:
me too - I thought it would be a lot of work!

Reply:
@DilwoarH it doesn't look like we resolved this?

Reply:
Hey @kevindew - what does `.each.with_index(1)` do?

Reply:
The above doesn't work, this is what I've come up:

```ruby
...
begin
  reordered_subsections = []
  params[:section_order_save].each do |id, position|
    reordered_subsections << { id: id, position: Integer(position) }
  end
rescue ArgumentError, TypeError
  return "Invalid request"
end

reordered_subsections
.sort_by { |section| section[:value] }
.each.with_index(1) do |section, index|
  sub_section = page.sub_sections.find(section[:id])
  sub_section.update_column(:position, index)
end
```